### PR TITLE
feat(1958): Hapi v19 dependencies upgrade. BREAKING CHANGE: v19.x.x

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Screwdriver hapi.js dependencies are outdated and not supported anymore.

## Objective

This PR upgrades hapi and node fixes all schema related changes to be compatible with latest @hapi js dependencies.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1958

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,13 +48,13 @@
...
     "clone": "^2.1.2",
-    "hoek": "^5.0.4",
-    "joi": "^13.7.0",
+    "joi": "^17.1.1",
     "js-yaml": "^3.13.0",
     "keymbinatorial": "^1.1.5",
-    "screwdriver-data-schema": "^19.1.1",
-    "screwdriver-workflow-parser": "^2.0.0",
+    "screwdriver-data-schema": "git://github.com/screwdriver-cd/data-schema.git#hapi-v19",
+    "screwdriver-workflow-parser": "git://github.com/screwdriver-cd/workflow-parser.git#hapi-v19",
```
Comment (tkyi) [inline package.json]: reminder to update before merge

File: test/index.test.js [hunk 2]
```diff
@@ -124,7 +124,7 @@ describe('config parser', () => {
                 () => parser(loadData('bad-step-name.yaml'))
                     .then((data) => {
                         assert.match(data.jobs.main[0].commands[0].command,
-                            /"foo bar" only supports the following characters A-Z,a-z,0-9,-,_/);
+                            /foo bar" only supports the following characters A-Z,a-z,0-9,-,_/);
```
Comment (tkyi) [inline test/index.test.js]: is this weird? why is it missing a leading quotation mark?

Comment (pritamstyz4ever) [reply test/index.test.js]: The complete message is actually [Validation Error] "jobs.main.steps[0].foo bar" ..., so we are matching the pattern.

File: lib/phase/structural.js [hunk 3]
```diff
@@ -49,22 +48,22 @@ function checkAdditionalRules(data) {
...
+        if (errors) {
+            throw errors;
+        }
 
-            return resolve(data);
-        });
-    })
-);
+        return data;
+    } catch (err) {
```
Comment (jithine) [inline lib/phase/structural.js]: why is this catch needed ? can't the entire `try..catch` be omitted and let the function throw err

File: test/index.test.js (hunk 2 above)
Comment (jithine) [reply test/index.test.js]: shouldn't it have matched even with the `"`?

Comment (pritamstyz4ever) [reply test/index.test.js]: Actual error message now includes the entire object structure.
ValidationError: "jobs.main.steps[0].foo bar" only supports the following characters A-Z,a-z,0-9,-,_
which is not a correct regex in itself. So made a limited change to message.

Comment (jithine) [reply test/index.test.js]: missed that

**Review by jithine — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**